src/main/scala/scalaz/DistributedModule.scala

```diff
@@ -0,0 +1,58 @@
+package scalaz
+
+trait Member
+
+sealed trait Membership
+object Membership {
+  final case class Join(member: Member) extends Membership
+  final case class Leave(member: Member) extends Membership
+  final case class Unreachable(member: Member) extends Membership
+}
+
+sealed trait ValueChanged[A]
+object ValueChanged {
+  final case class Modify[A](member: Member, old: A, newV: A) extends ValueChanged[A]
+  final case class Add[A](member: Member, newV: A) extends ValueChanged[A]
+  final case class Remove[A](member: Member, value: A) extends ValueChanged[A]
+}
+
+trait DistributedModule[F[_]] {
+  type Type[A]
+  type Prism[A, B]
+  type Traversal[A, B]
+
+  sealed trait PathElem[A, B]
```

Comment:
Could you briefly explain the intention? As far as I can see, this hierarchy takes place of lenses present in the gist.